### ISSUE TITLE
Filtered child entities in projection are not tracked

### ISSUE THREAD

**Issue by havotto** [aspnet/EntityFrameworkCore#10124]
When child entities are filtered in a projection, these child entities are not tracked, but added to the parent's navigation collection. Also, the inverse navigation properties are not fixed up.

This issue is quite related to [issue 7131](https://github.com/aspnet/EntityFrameworkCore/issues/7131)

### Steps to reproduce
Here is the repo with tests for details: [Repo](https://bitbucket.org/h_otto/efcore2entitytrackingbug.git)

### Further technical details
EF Core version: 2.0.0
Database Provider: Microsoft.EntityFrameworkCore.Sqlite
Operating system: Win 10
IDE: Visual Studio 2017

**Comment by ajcvickers** [comment]
Note for triage:
In this case, Child entities are tracked:
```C#
var result = ctx.Parents.Select(p =>
new
{
    Parent = p,
    Children = p.Children
})
.First();
```

In this case, Child entities are not tracked:
```C#
var result = ctx.Parents.Select(p =>
new
{
    Parent = p,
    Children = p.Children.Where(c => c.Name.Length > 0).ToList()
})
.First();
```

**Comment by anpete** [comment]
Similar to #7569. @maumar is there a dupe of this?

**Comment by maumar** [comment]
fixed in b95f23ffa4fbba305b3f0ea4c864fe53f456ad58
